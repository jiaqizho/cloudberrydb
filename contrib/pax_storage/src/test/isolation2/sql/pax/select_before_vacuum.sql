-- @Description Ensures that a select before a vacuum operation is ok
-- 
DROP TABLE IF EXISTS pax_tbl;
CREATE TABLE pax_tbl (a INT);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);

DELETE FROM pax_tbl WHERE a < 128;
1: BEGIN;
1: SELECT COUNT(*) FROM pax_tbl;
2&: VACUUM FULL pax_tbl;
1: COMMIT;
2<:
1: SELECT COUNT(*) FROM pax_tbl;
3: INSERT INTO pax_tbl VALUES (0);
