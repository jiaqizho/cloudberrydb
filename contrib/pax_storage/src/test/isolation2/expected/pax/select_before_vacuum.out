-- @Description Ensures that a select before a vacuum operation is ok
--
DROP TABLE IF EXISTS pax_tbl;
DROP
CREATE TABLE pax_tbl (a INT);
CREATE
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000

DELETE FROM pax_tbl WHERE a < 128;
DELETE 2667
1: BEGIN;
BEGIN
1: SELECT COUNT(*) FROM pax_tbl;
 count 
-------
 18333 
(1 row)
2&: VACUUM FULL pax_tbl;  <waiting ...>
1: COMMIT;
COMMIT
2<:  <... completed>
VACUUM
1: SELECT COUNT(*) FROM pax_tbl;
 count 
-------
 18333 
(1 row)
3: INSERT INTO pax_tbl VALUES (0);
INSERT 1
